### PR TITLE
CTCTRADERS-186: Type ahead changes for transport nationality

### PR DESCRIPTION
**Pull request by ashwinirudrappa**

### CODE REVIEW

File: test/controllers/events/transhipments/TransportNationalityControllerSpec.scala [hunk 1]
```diff
@@ -197,4 +162,43 @@ class TransportNationalityControllerSpec extends SpecBase with MockitoSugar with
...
+      "mrn"       -> mrn,
+      "mode"      -> NormalMode,
+      "countries" -> countriesJson(preSelect)
+    )
+
+    templateCaptor.getValue mustEqual transportNationalityTemplate
+    jsonCaptor.getValue must containJson(expectedJson)
+
+    application.stop()
+  }
```
Comment (garethgooch) [inline test/controllers/events/transhipments/TransportNationalityControllerSpec.scala]: Discussion point about bringing this back into the test itself, in this form you have to search to find what the test is doing and it looks like the first test and second are the same

File: test/forms/events/transhipments/TransportNationalityFormProviderSpec.scala [hunk 2]
```diff
@@ -18,15 +18,15 @@ package forms.events.transhipments
...
 class TransportNationalityFormProviderSpec extends StringFieldBehaviours {
 
-  val requiredKey = "transportNationality.error.required"
-  val lengthKey   = "transportNationality.error.length"
-  val maxLength   = VehicularTranshipment.Constants.transportCountryLength
-
-  val form = new TransportNationalityFormProvider()()
+  val requiredKey    = "transportNationality.error.required"
+  val lengthKey      = "transportNationality.error.length"
+  val maxLength: Int = VehicularTranshipment.Constants.transportCountryLength
```
Comment (garethgooch) [inline test/forms/events/transhipments/TransportNationalityFormProviderSpec.scala]: random type annotation?

File: test/forms/events/transhipments/TransportNationalityFormProviderSpec.scala [hunk 3]
```diff
@@ -18,15 +18,15 @@ package forms.events.transhipments
...
+import models.reference.Country
 import play.api.data.FormError
 
 class TransportNationalityFormProviderSpec extends StringFieldBehaviours {
 
-  val requiredKey = "transportNationality.error.required"
-  val lengthKey   = "transportNationality.error.length"
-  val maxLength   = VehicularTranshipment.Constants.transportCountryLength
-
-  val form = new TransportNationalityFormProvider()()
```
Comment (garethgooch) [inline test/forms/events/transhipments/TransportNationalityFormProviderSpec.scala]: We seem to be extending StringField behaviours on this and testing that this field binds with a string, but isn't this now a country and should only bind a valid country?

File: test/forms/events/transhipments/TransportNationalityFormProviderSpec.scala [hunk 4]
```diff
@@ -38,17 +38,25 @@ class TransportNationalityFormProviderSpec extends StringFieldBehaviours {
       stringsWithMaxLength(maxLength)
     )
 
-    behave like fieldWithMaxLength(
-      form,
-      fieldName,
-      maxLength   = maxLength,
-      lengthError = FormError(fieldName, lengthKey, Seq(maxLength))
-    )
```
Comment (garethgooch) [inline test/forms/events/transhipments/TransportNationalityFormProviderSpec.scala]: If we are a string this should be back?

File: test/forms/events/transhipments/TransportNationalityFormProviderSpec.scala [hunk 5]
```diff
@@ -38,17 +38,25 @@ class TransportNationalityFormProviderSpec extends StringFieldBehaviours {
...
+      val field     = boundForm("value")
+      field.errors mustNot be(empty)
+    }
+
+    "bind a country code which is in the list" in {
+
+      val boundForm = form.bind(Map("value" -> "AD"))
+      val field     = boundForm("value")
+      field.errors must be(empty)
+    }
```
Comment (garethgooch) [inline test/forms/events/transhipments/TransportNationalityFormProviderSpec.scala]: maybe move this into behaviours and we can use this on the pages

File: test/controllers/events/transhipments/TransportNationalityControllerSpec.scala (hunk 1 above)
Comment (ashwinirudrappa) [reply test/controllers/events/transhipments/TransportNationalityControllerSpec.scala]: @garethgooch Yes its doing the same thing, thats the reason its moved to common block, the only separation is the parameter that is passed

File: test/forms/events/transhipments/TransportNationalityFormProviderSpec.scala (hunk 2 above)
Comment (ashwinirudrappa) [reply test/forms/events/transhipments/TransportNationalityFormProviderSpec.scala]: @garethgooch ???

File: app/controllers/events/transhipments/TransportNationalityController.scala [hunk 6]
```diff
@@ -16,75 +16,90 @@
...
+      "form"      -> form,
+      "mrn"       -> mrn,
+      "mode"      -> mode,
+      "countries" -> countryJsonList(form.value, countries)
+    )
+
+    renderer.render("events/transhipments/transportNationality.njk", json).map(status(_))
+  }
+
+  private def countryJsonList(value: Option[Country], countries: Seq[Country]): Seq[JsObject] = {
```
Comment (amaalali) [inline app/controllers/events/transhipments/TransportNationalityController.scala]: Probably a refactor this and same logic in `EventCountryController`

File: test/controllers/events/transhipments/TransportNationalityControllerSpec.scala [hunk 7]
```diff
@@ -39,75 +42,33 @@ import scala.concurrent.Future
...
+  val form: Form[Country]      = formProvider(countries)
 
-  lazy val transportNationalityRoute       = routes.TransportNationalityController.onPageLoad(mrn, eventIndex, NormalMode).url
-  private val transportNationalityTemplate = "events/transhipments/transportNationality.njk"
+  val mockReferenceDataConnector: ReferenceDataConnector = mock[ReferenceDataConnector]
+  lazy val transportNationalityRoute: String             = routes.TransportNationalityController.onPageLoad(mrn, eventIndex, NormalMode).url
+  private val transportNationalityTemplate               = "events/transhipments/transportNationality.njk"
+
+  private val templateCaptor: ArgumentCaptor[String] = ArgumentCaptor.forClass(classOf[String])
+  private val jsonCaptor: ArgumentCaptor[JsObject]   = ArgumentCaptor.forClass(classOf[JsObject])
```
Comment (amaalali) [inline test/controllers/events/transhipments/TransportNationalityControllerSpec.scala]: Can we move the argument captors to the scope of the test method

**Pull request merged**